### PR TITLE
Implement LIST QUERIES that lists all running queries

### PR DESCRIPTION
**Pull request by wladekb**
Since now we've had a painful lack of visibility into what's going on under the hood of influxdb.

I'm now adding LIST QUERIES that lists all running queries along with the time they're already being executed. Hopefully this will help us in a few cases.

Note: this is already live on our servers so you can give it a try..

/cc @macbre @jcellary @owend @drsnyder @ArturKlajnerok

### CODE REVIEW

**Comment by macbre** [comment]
Let's give it a Go ;)

File: coordinator/coordinator.go [hunk 1]
```diff
@@ -16,11 +18,93 @@ import (
...
+
+	return runningQuery
+}
+
+func NewRunningQueries() *RunningQueries {
+	runningQueries := &RunningQueries{
+		data:                 make(map[*RunningQuery]struct {}),
+	}
+
+	return runningQueries
```
Comment (ArturKlajnerok) [inline coordinator/coordinator.go:59]: You can return directly `&RunningQueries{...}` no need to assign `runningQueries` :smile_cat:

File: coordinator/coordinator.go [hunk 2]
```diff
@@ -16,11 +18,93 @@ import (
...
+	userName             string
+	databaseName         string
+	queryString          string
+	startTime            time.Time
+}
+
+type RunningQueryList  []*RunningQuery
+type RunningQueries struct {
+	data                 map[*RunningQuery]struct{}
+	lock                 sync.Mutex
```
Comment (ArturKlajnerok) [inline coordinator/coordinator.go]: You can `Mutex` use as part of struct, without assigning to specific field.

File: coordinator/coordinator.go [hunk 3]
```diff
@@ -16,11 +18,93 @@ import (
...
+		data:                 make(map[*RunningQuery]struct {}),
+	}
+
+	return runningQueries
+}
+
+func (self *RunningQueries) Add(q *RunningQuery) {
+	self.lock.Lock()
+	self.data[q] = struct{}{}
+	self.lock.Unlock()
```
Comment (ArturKlajnerok) [inline coordinator/coordinator.go]: If you like it's possible to keep `Lock` and `Unlock` one after another with using `defer`

File: coordinator/coordinator.go [hunk 4]
```diff
@@ -16,11 +18,93 @@ import (
...
+
+func (self *RunningQueries) Remove(q *RunningQuery) {
+	self.lock.Lock()
+	delete(self.data, q)
+	self.lock.Unlock()
+}
+
+func (self *RunningQueries) All() <-chan *RunningQuery {
+	ch := make(chan *RunningQuery)
+	self.lock.Lock()
```
Comment (ArturKlajnerok) [inline coordinator/coordinator.go]: Maybe it would be more safe/consistent to move `Lock()` inside the goroutine ?

**Comment by ArturKlajnerok** [comment]
I don't know well enough influxdb implementation details, that those changes are affecting but other than that LGTM :+1:

File: coordinator/coordinator.go (hunk 1 above)
Comment (wladekb) [inline coordinator/coordinator.go:59]: I followed what it looks like a "pattern". 
See eg. NewCoordinator here: https://github.com/Wikia/influxdb/pull/1/files#diff-cd24963c47c3445deb5ec946ac538602R110

File: coordinator/coordinator.go (hunk 4 above)
Comment (wladekb) [inline coordinator/coordinator.go]: Good catch. Thanks.

**Pull request merged**